src/multipliers/VotingStreakMultiplier.sol

```diff
@@ -0,0 +1,113 @@
+import {IMultiplier} from "src/interfaces/multipliers/IMultiplier.sol";
+import {YieldDistributor} from "src/YieldDistributor.sol";
+import {IVotingStreakMultiplier} from "src/interfaces/multipliers/IVotingStreakMultiplier.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+/// @title VotingStreakMultiplier
+/// @notice A contract for managing voting streak multipliers
+/// @dev Implements IMultiplier and IVotingStreakMultiplier interfaces
+contract VotingStreakMultiplier is Initializable, OwnableUpgradeable, IMultiplier, IVotingStreakMultiplier {
+    /// @notice The maximum multiplier incrementation
+    uint256 public maxMultiplier;
+
+    /// @notice The increment value for the multiplier
+    uint256 public multiplierIncrement;
+
+    /// @notice The YieldDistributor contract
+    YieldDistributor public yieldDistributor;
+
+    /// @notice Mapping of user addresses to their current multiplier factor
+    mapping(address => uint256) public userToMultiplier;
+
+    /// @notice Mapping of user addresses to their multiplier validity period
+    mapping(address => uint256) public userToValidity;
+
+    /// @notice Emitted when a user's multiplier is updated
+    /// @param user The address of the user
+    /// @param newFactor The new multiplier factor
+    /// @param validity The new validity period
+    event MultiplierUpdated(address indexed user, uint256 newFactor, uint256 validity);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initializes the contract
+    /// @param _yieldDistributor The address of the YieldDistributor contract
+    /// @param _multiplierIncrement The initial multiplier increment value
+    /// @param _maxMultiplier The maximum multiplier value
+    function initialize(address _yieldDistributor, uint256 _multiplierIncrement, uint256 _maxMultiplier)
+        public
+        initializer
+    {
+        __Ownable_init(msg.sender);
+
+        yieldDistributor = YieldDistributor(_yieldDistributor);
+        multiplierIncrement = _multiplierIncrement;
+        maxMultiplier = _maxMultiplier;
+    }
+
+    /// @notice Updates the user's multiplier when a vote is cast
+    /// @param voter The address of the voter
+    function onVoteCast(address voter) external {
+        require(msg.sender == address(yieldDistributor), "Only YieldDistributor can call");
+
+        uint256 currentMultiplier = _getMultiplyingFactor(voter);
+        uint256 newMultiplier = (currentMultiplier == 0)
+            ? multiplierIncrement
+            : Math.min(currentMultiplier + multiplierIncrement, maxMultiplier * multiplierIncrement);
+
+        userToMultiplier[voter] = newMultiplier;
+        userToValidity[voter] = yieldDistributor.lastClaimedBlockNumber() + 2 * yieldDistributor.cycleLength();
+
+        emit MultiplierUpdated(voter, newMultiplier, userToValidity[voter]);
```

Comment:
I don't see how we're keeping track of the streak here 

note that there is already a `accountToLastVoted` variable in the YD that can be used to see whether the user has broken their streak or not 

also i think we can change the `getMultiplyingFactor` to not be a view so this logic an be put in there, that way we don't need to introduce `onVoteCast` to the YD